frontend/src/hooks/use-pagination/use-pagination.hook.ts

```diff
@@ -0,0 +1,32 @@
+import { PaginationDefaultValue } from 'common/enums/enums';
+import { useSearchParams, useState } from 'hooks/hooks';
+
+type UsePaginationResult = {
+  page: number;
+  handlePageChange: (page: number) => void;
+};
+
+const usePagination = (): UsePaginationResult => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const getValidPageFromSearchParams = (): number => {
+    const pageFromQuery = Number(searchParams.get('page'));
+
+    return pageFromQuery > 0
+      ? pageFromQuery
+      : PaginationDefaultValue.DEFAULT_PAGE;
+  };
+
+  const [page, setPage] = useState<number>(getValidPageFromSearchParams());
+
+  const handlePageChange = (page: number): void => {
+    const updatedSearchParams = new URLSearchParams(searchParams.toString());
+    updatedSearchParams.set('page', String(page));
```

Comment:
same

Reply:
Did I understand you correctly, that we need to pass `pageFromQuery`, `searchParams`, `setSearchParams` through hook params?

Reply:
we need to pass ths string `page`. since we can have more that 1 table on the page